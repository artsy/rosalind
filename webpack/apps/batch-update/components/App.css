#root {
  position: absolute;
  left: 0;
  top: 50px;
  right: 0;
  bottom: 0;
}

/* basic layout */

.App {
  min-height: calc(100vh - 50px);
  display: flex;
  flex-flow: row wrap;
  align-content: flex-start;
  align-items: stretch;
}

.SearchForm, .SearchResults {
  padding: 1em;
}

.SearchForm {
  flex-grow: 1;
  flex-basis: 150px;
  border-right: solid 1px #eee;
  margin-top: ;
}

.SearchResults {
  position: relative;
  flex-grow: 5;
  flex-basis: 400px;
}

/* components */

.SearchForm-SectionHeader {
  color: #999;
  text-transform: uppercase;
  letter-spacing: 0.1em;
  font-size: 0.75em;
  font-family:'ITC Avant Garde Gothic W01';
  font-weight: 400;
  margin-top: 1.5em;
}

.SelectedGene,
.SelectedTag,
.SelectedPartner, 
.SelectedFair {
  font-weight: bold;
  margin: 0.25em 0;
}

.SelectedGene-remove,
.SelectedTag-remove,
.SelectedPartner-remove, 
.SelectedFair-remove {
  display: inline-block;
  height: 2em;
  width: 2em;
  line-height: 2em;
  text-align: center;
  color: #ddd;
  margin-left: 0.5em;
  border-radius: 1em;
  text-decoration: none;
}

.SelectedGene-remove:hover,
.SelectedTag-remove:hover,
.SelectedPartner-remove:hover, 
.SelectedFair-remove:hover {
  color: #333;
  cursor: pointer;
  background: #eee;
}

.FilterOptions {
  font-size: 80%;
  color: #999;
  line-height: 140%;
  margin-top: 2em;
  opacity: 0.25;
  transition: opacity 0.75s;
}

.FilterOptions:hover {
  opacity: 1;
  transition: opacity 0.25s;
}

.PublishedFilter,
.DeletedFilter,
.GenomedFilter {
  margin-top: 1em;
}

.PublishedFilter a,
.DeletedFilter a,
.GenomedFilter a {
  color: #ccc;
  margin-right: 0.5em;
}

.PublishedFilter a.active,
.DeletedFilter a.active,
.GenomedFilter a.active {
  color: #333;
  font-weight: bold;
  text-decoration: none;
  cursor: default;;
}

.SearchResults {
  display: flex;
  flex-flow: row wrap;
  align-content: flex-start;
}

.ArtworkResult {
  /*outline: solid 1px red;*/
  min-width: 115px;
  max-width: calc(20% - 1.5em);
  margin: 0.75em;
  flex: 1 1 calc(20% - 1.5em);
}

.ArtworkResult img {
  width: 100%;
}

.ArtworkResult figcaption {
  position: relative;
  font-size: 80%;
  max-height: 3.75em;
  line-height: 1.25em;
  overflow: hidden;
}

.ArtworkPreview-Overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: hsla(0, 0%, 0%, 0.5);
  z-index: 1;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.ArtworkPreview {
  /*flex: 0 0 66%;*/
  box-sizing: border-box;
  padding: 2em;
  max-height: 90%;
  z-index: 2;
  background: white;
  box-shadow: 0 0 20px hsla(0, 0%, 0%, 0.5);

  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.ArtworkPreview-Image {
  flex: 0 1 40%;
}

.ArtworkPreview-Image img {
  /*min-width: 500px;*/
}

.ArtworkPreview-Details {
  max-width: 20em;
  margin-left: 1em;
  flex: 0 1 60%;
}

.ArtworkPreview-Name {
  font-weight: bold;
}

.ArtworkPreview-Status {
  font-size: 0.9em;
  line-height: 125%;
  margin: 1em 0;
}

.ArtworkPreview-Links a {
  margin: 0 0.25em;
}
