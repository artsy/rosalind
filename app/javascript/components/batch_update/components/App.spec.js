import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'
import App from './App'

it('renders correctly', () => {
  const app = renderer.create(<App />)
  const tree = app.toJSON()
  expect(tree).toMatchSnapshot()
})

let app

beforeEach(() => {
  const fetchArtworksPromise = new Promise(() => {})
  window.fetch = jest.fn(() => fetchArtworksPromise)

  const wrapper = shallow(<App />)
  app = wrapper.instance()
})

describe('state mutations', () => {
  describe('search criteria', () => {
    beforeAll(() => {
      // We assume the existence of a <meta> tag containing the
      // Rails-generated CSRF token. This is how we mock access to
      // that tag.
      // https://github.com/facebook/jest/issues/2297
      Object.defineProperty(document, 'querySelector', {
        value: () => {
          return { content: 'very secret csrf token' }
        },
      })
    })

    test('genes can be added and removed', () => {
      const gene1 = { id: 'one', name: 'One' }
      const gene2 = { id: 'two', name: 'Two' }

      app.onAddGene(gene1)
      app.onAddGene(gene2)
      expect(app.state).toMatchObject({
        genes: [gene1, gene2],
      })

      app.onRemoveGene('One')
      expect(app.state).toMatchObject({
        genes: [gene2],
      })
    })

    test('tags can be added and removed', () => {
      const tag1 = { id: 'one', name: 'One' }
      const tag2 = { id: 'two', name: 'Two' }

      app.onAddTag(tag1)
      app.onAddTag(tag2)
      expect(app.state).toMatchObject({
        tags: [tag1, tag2],
      })

      app.onRemoveTag('One')
      expect(app.state).toMatchObject({
        tags: [tag2],
      })
    })

    test('artists can be added and removed', () => {
      const artist1 = { id: 'abc123', name: 'Alice', slug: 'alice' }
      const artist2 = { id: 'def456', name: 'Bob', slug: 'bob' }

      app.onAddArtist(artist1)
      app.onAddArtist(artist2)
      expect(app.state).toMatchObject({
        artists: [artist1, artist2],
      })

      app.onRemoveArtist('abc123')
      expect(app.state).toMatchObject({
        artists: [artist2],
      })
    })

    test('keywords can be added and removed', () => {
      const keyword1 = 'foo'
      const keyword2 = 'bar'

      app.onAddKeyword(keyword1)
      app.onAddKeyword(keyword2)
      expect(app.state).toMatchObject({
        keywords: [keyword1, keyword2],
      })

      app.onRemoveKeyword('foo')
      expect(app.state).toMatchObject({
        keywords: [keyword2],
      })
    })

    test('arbitrary state keys can be set and cleared', () => {
      app.updateStateFor('foo', 'bar')
      expect(app.state).toMatchObject({
        foo: 'bar',
      })

      app.clearStateFor(null, 'foo')
      expect(app.state).toMatchObject({
        foo: null,
      })
    })
  })

  describe('artwork selection', () => {
    test('individual artworks can be toggled', () => {
      const work = { id: 'guernica', name: 'Guernica' }

      app.onToggleArtwork(work)
      expect(app.state).toMatchObject({
        selectedArtworkIds: ['guernica'],
      })

      app.onToggleArtwork(work)
      expect(app.state).toMatchObject({
        selectedArtworkIds: [],
      })
    })

    test('the full list of artworks can be selected and deselected', () => {
      const work1 = { id: 'guernica', name: 'Guernica' }
      const work2 = { id: 'soup-can', name: 'Soup Can' }
      app.setState({ artworks: [work1, work2] })

      app.onSelectAllArtworks()
      expect(app.state).toMatchObject({
        selectedArtworkIds: ['guernica', 'soup-can'],
      })

      app.onDeselectAllArtworks()
      expect(app.state).toMatchObject({
        selectedArtworkIds: [],
      })
    })
  })

  describe('artwork preview', () => {
    const work1 = { id: 'guernica', name: 'Guernica' }
    const work2 = { id: 'soup-can', name: 'Soup Can' }
    const work3 = { id: 'shark', name: 'Shark' }

    beforeEach(() => {
      app.setState({ artworks: [work1, work2, work3] })
    })

    test('an artwork can be selected for preview', () => {
      app.onPreviewArtwork(work2)
      expect(app.state).toMatchObject({
        previewedArtwork: work2,
      })
    })

    test('the previous artwork can be selected for preview', () => {
      app.onPreviewArtwork(work2)
      app.onPreviewPrevious()
      expect(app.state).toMatchObject({
        previewedArtwork: work1,
      })
    })

    test('the next work can be selected for preview', () => {
      app.onPreviewArtwork(work2)
      app.onPreviewNext()
      expect(app.state).toMatchObject({
        previewedArtwork: work3,
      })
    })
  })
})

describe('getCommonGenes', () => {
  let commonGenes

  beforeEach(() => {
    app.setState({
      artworks: [
        {
          id: 'gold-sculpture',
          genes: ['Art', 'Sculpture', 'Gold'],
        },
        {
          id: 'silver-sculpture',
          genes: ['Art', 'Sculpture', 'Silver'],
        },
        {
          id: 'gold-jewelry',
          genes: ['Art', 'Jewelry', 'Gold'],
        },
      ],
      selectedArtworkIds: ['gold-sculpture', 'silver-sculpture'],
    })

    commonGenes = app.getCommonGenes()
  })

  it('returns genes shared by all selected artworks', () => {
    expect(commonGenes).toEqual(['Sculpture'])
  })

  it('excludes some overly common genes', () => {
    expect(commonGenes).not.toContain('Art')
  })
})
