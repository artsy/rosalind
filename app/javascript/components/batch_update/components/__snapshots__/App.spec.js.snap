// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  min-height: calc(100vh - 50px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 150px;
  -ms-flex-preferred-size: 150px;
  flex-basis: 150px;
  border-bottom: solid 1px #eee;
  padding: 1em;
}

.c11 {
  -webkit-box-flex: 5;
  -webkit-flex-grow: 5;
  -ms-flex-positive: 5;
  flex-grow: 5;
  -webkit-flex-basis: 400px;
  -ms-flex-preferred-size: 400px;
  flex-basis: 400px;
  padding: 1em;
}

.c2 h2 {
  font-family: 'ITC Avant Garde Gothic W01';
  color: #999;
  font-size: 0.75em;
  font-weight: 400;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  margin-top: 1.5em;
  text-transform: uppercase;
}

.c4 input {
  width: 100%;
  margin: 1em 0 0 0;
  line-height: 2em;
  font-size: 1em;
  border: none;
  border-bottom: solid 1px #ddd;
  background: none;
}

.c4 input:focus {
  outline: none;
}

.c3 input {
  width: 100%;
  margin: 1em 0 0 0;
  line-height: 2em;
  font-size: 1em;
  border: none;
  border-bottom: solid 1px #ddd;
  background: none;
}

.c3 input:focus {
  outline: none;
}

.c8 {
  font-size: 80%;
  color: #333;
  line-height: 140%;
  margin: 2em 0;
  opacity: 0.5;
  -webkit-transition: opacity 0.5s;
  transition: opacity 0.5s;
}

.c8:hover {
  opacity: 1;
  -webkit-transition: opacity 0.5s;
  transition: opacity 0.5s;
}

.c8 .filter {
  margin-top: 1em;
}

.c8 .filter a {
  color: #999;
  margin-right: 0.5em;
}

.c8 .filter a.active {
  color: #333;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: default;
}

.c5 {
  margin-top: 8px;
}

.c7 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 16px;
  display: inline;
  margin-right: 0.5px;
}

.c10 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  padding-top: 1px;
}

.c18 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 28px;
  line-height: 36px;
  margin-top: 16px;
  margin-bottom: 4px;
}

.c20 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 28px;
  line-height: 36px;
  margin-top: 32px;
  margin-bottom: 4px;
}

.c9 {
  cursor: pointer;
  position: relative;
  white-space: nowrap;
  border: 2px solid;
  border-radius: 3px;
  border-radius: 3px;
  padding-left: 20px;
  padding-right: 20px;
  width: 100%;
  height: 41px;
  -webkit-transition: 0.25s ease;
  transition: 0.25s ease;
  background-color: #000;
  border-color: #000;
  color: #FFF;
}

.c9.loading {
  -webkit-transition: none;
  transition: none;
  background-color: transparent;
  color: transparent;
  border: 0;
  cursor: auto;
}

.c9.disabled {
  background-color: #E5E5E5;
  border-color: #E5E5E5;
  color: #FFF;
  pointer-events: none;
}

.c17 {
  cursor: pointer;
  position: relative;
  white-space: nowrap;
  border: 2px solid;
  border-radius: 3px;
  border-radius: 3px;
  padding-left: 20px;
  padding-right: 20px;
  height: 41px;
  -webkit-transition: 0.25s ease;
  transition: 0.25s ease;
  background-color: #000;
  border-color: #000;
  color: #FFF;
}

.c17.loading {
  -webkit-transition: none;
  transition: none;
  background-color: transparent;
  color: transparent;
  border: 0;
  cursor: auto;
}

.c17.disabled {
  background-color: #E5E5E5;
  border-color: #E5E5E5;
  color: #FFF;
  pointer-events: none;
}

.c23 {
  cursor: pointer;
  position: relative;
  white-space: nowrap;
  border: 2px solid;
  border-radius: 3px;
  border-radius: 3px;
  margin-left: 10px;
  margin-right: 10px;
  padding-left: 20px;
  padding-right: 20px;
  height: 41px;
  -webkit-transition: 0.25s ease;
  transition: 0.25s ease;
  background-color: #FFF;
  border-color: #E5E5E5;
  color: #000;
}

.c23.loading {
  -webkit-transition: none;
  transition: none;
  background-color: transparent;
  color: transparent;
  border: 0;
  cursor: auto;
}

.c23.disabled {
  background-color: #E5E5E5;
  border-color: #E5E5E5;
  color: #FFF;
  pointer-events: none;
}

.c6 {
  position: relative;
}

.c6 label {
  padding: 0;
  margin: 0;
}

.c6 select {
  background: none;
  border: none;
  cursor: pointer;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  text-indent: 0.01px;
  text-overflow: "";
  background-color: #E5E5E5;
  border-radius: 2px;
  padding: 5px 26px 5px 10px;
  margin: 0;
}

.c6 select:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #000;
}

.c6 select option:not(:checked) {
  color: black;
}

.c6 select:hover {
  background-color: #C2C2C2;
}

.c6 select:focus {
  border-color: #6E1EFF;
}

.c6::after {
  border-left: 4px solid transparent;
  border-right: 4px solid transparent;
  border-top: 4px solid #000;
  width: 0;
  height: 0;
  content: "";
  cursor: pointer;
  margin-left: -18px;
  pointer-events: none;
  position: absolute;
  top: 10px;
}

.c16 {
  color: #666666;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 .counts {
  display: inline-block;
  width: 75%;
  padding: 0.75em;
}

.c12 .select {
  display: inline-block;
  width: 25%;
  padding: 0.75em;
  text-align: right;
}

.c12 .select a {
  margin: 0 0.25em;
}

.c12 .results {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c21 {
  position: fixed;
  left: 50%;
  top: 40%;
  width: 600px;
  height: auto;
  background: white;
  z-index: 2;
  padding: 20px;
  visibility: hidden;
  opacity: 0;
  -webkit-transform: scale(0.95) translate(-50%,-50%);
  -ms-transform: scale(0.95) translate(-50%,-50%);
  transform: scale(0.95) translate(-50%,-50%);
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.c21.modal-open {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1) translate(-50%,-50%);
  -ms-transform: scale(1) translate(-50%,-50%);
  transform: scale(1) translate(-50%,-50%);
  -webkit-transition: all 0.125s;
  transition: all 0.125s;
}

.c21 h1 {
  font-weight: normal;
  font-size: 1.5em;
  color: #666666;
}

.c21 section {
  color: #333333;
  border-top: solid 1px #E5E5E5;
  margin-top: 1em;
  padding-top: 1em;
}

.c21 section p {
  margin-bottom: 1em;
}

.c22 {
  border-top: solid 1px #E5E5E5;
  margin-top: 1em;
  padding-top: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c22 a {
  margin-left: 1em;
}

.c14 {
  padding: 18px 30px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 20px;
  border-bottom: solid 1px #DBDBDB;
}

.c19 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  text-align: center;
}

.c13 {
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;
  background: white;
  overflow: scroll;
  visibility: hidden;
  opacity: 0;
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.c13.modal-open {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 0.125s;
  transition: all 0.125s;
}

.c24 {
  position: fixed;
  top: 0;
  right: 0;
  width: 25%;
  min-width: 20em;
  padding: 1em;
  font-size: 0.9em;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c1 {
    border-bottom: none;
    border-right: solid 1px #eee;
  }
}

@media not all and (pointer:coarse),not all and (-moz-touch-enabled:1) {
  .c9:hover {
    background-color: #6E1EFF;
    border-color: #6E1EFF;
    color: #FFF;
  }
}

@media not all and (pointer:coarse),not all and (-moz-touch-enabled:1) {
  .c17:hover {
    background-color: #6E1EFF;
    border-color: #6E1EFF;
    color: #FFF;
  }
}

@media not all and (pointer:coarse),not all and (-moz-touch-enabled:1) {
  .c23:hover {
    background-color: #FFF;
    border-color: #000;
    color: #000;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className=""
    >
      <div
        className="c2"
      />
      <div
        className="c3"
      >
        <input
          onKeyUp={[Function]}
          placeholder="Add a keyword"
          type="text"
        />
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-gene-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-gene-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Add a gene"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-gene-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-tag-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-tag-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Add a tag"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-tag-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-artist-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-artist-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Add an artist"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-artist-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-partner-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-partner-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Select a partner"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-partner-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-fair-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-fair-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Select a fair"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-fair-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-attribution-class-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-attribution-class-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Select an attribution class"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-attribution-class-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="c4"
      >
        <input
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Created after date"
          type="text"
          value=""
        />
        <div
          className="parsed"
        >
          <a
            href="#"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="c4"
      >
        <input
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Created before date"
          type="text"
          value=""
        />
        <div
          className="parsed"
        >
          <a
            href="#"
            onClick={[Function]}
          />
        </div>
      </div>
      <div>
        <div
          className="c3 minPriceInput"
        >
          <input
            onKeyUp={[Function]}
            placeholder="Minimum Price"
            type="text"
          />
        </div>
        <div
          className="c3 maxPriceInput"
        >
          <input
            onKeyUp={[Function]}
            placeholder="Maximum Price"
            type="text"
          />
        </div>
      </div>
      <div
        className="c3"
      >
        <input
          onKeyUp={[Function]}
          placeholder="Restrict to artwork IDs"
          type="text"
        />
      </div>
      <div
        className="c5"
      >
        <div
          className="c6"
          onSelect={[Function]}
          selected="RECENTLY_PUBLISHED"
          title="Sort"
        >
          <label>
            <div
              className="c7 "
              display="inline"
              fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
              fontSize="12px"
              style={Object {}}
            >
              Sort
              :
            </div>
            <select
              onChange={[Function]}
              value="RECENTLY_PUBLISHED"
            >
              <option
                value="MERCHANDISABILITY"
              >
                Merchandisability
              </option>
              <option
                value="RECENTLY_PUBLISHED"
              >
                Recently published
              </option>
            </select>
          </label>
        </div>
      </div>
      <div
        className="c8"
      >
        <div
          className="filter"
        >
          <div>
            Published?
          </div>
          <a
            className="active"
            href="#"
            onClick={[Function]}
          >
            All
          </a>
          <a
            className={null}
            href="#"
            onClick={[Function]}
          >
            True
          </a>
          <a
            className={null}
            href="#"
            onClick={[Function]}
          >
            False
          </a>
        </div>
        <div
          className="filter"
        >
          <div>
            Acquireable Or Offerable?
          </div>
          <a
            className="active"
            href="#"
            onClick={[Function]}
          >
            All
          </a>
          <a
            className={null}
            href="#"
            onClick={[Function]}
          >
            True
          </a>
          <a
            className={null}
            href="#"
            onClick={[Function]}
          >
            False
          </a>
        </div>
        <div
          className="filter"
        >
          <div>
            For Sale?
          </div>
          <a
            className="active"
            href="#"
            onClick={[Function]}
          >
            All
          </a>
          <a
            className={null}
            href="#"
            onClick={[Function]}
          >
            True
          </a>
          <a
            className={null}
            href="#"
            onClick={[Function]}
          >
            False
          </a>
        </div>
      </div>
      <button
        className="c9 disabled"
        disabled={true}
        height="41px"
        onClick={[Function]}
        width={1}
      >
        <div
          className="c10 "
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
          style={Object {}}
        >
          Edit Artworks
        </div>
      </button>
    </div>
  </div>
  <div
    className="c11"
  >
    <div
      className="c12"
    >
      <div
        className="results"
      />
    </div>
  </div>
  <div
    className="c13"
  >
    <div
      className="c14"
    >
      <div
        className="c15"
      >
        <a
          className="c16 cancel"
          href="#"
          onClick={[Function]}
        >
          Cancel
        </a>
        <div>
          0
           works selected
        </div>
        <button
          className="c17 disabled queue"
          disabled={true}
          height="41px"
          onClick={[Function]}
        >
          <div
            className="c10 "
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize="14px"
            style={Object {}}
          >
            Queue changes
          </div>
        </button>
      </div>
      <div
        className="c18 "
        fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
        fontSize="28px"
        style={Object {}}
      >
        Tags
      </div>
      <div
        className="c19"
      >
        There aren’t any tags that describe all of your selected works
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-tag-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-tag-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Add a tag"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-tag-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="c20 "
        fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
        fontSize="28px"
        style={Object {}}
      >
        Genes
      </div>
      <div
        className="c19"
      >
        There aren’t any genes that describe all of your selected works
      </div>
      <div
        className="GenericAutosuggest"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-gene-autosuggest"
          className="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-gene-autosuggest"
            autoComplete="off"
            className="react-autosuggest__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Add a gene"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-gene-autosuggest"
            role="listbox"
          />
        </div>
      </div>
      <div
        className="c21"
      >
        <h1>
          Are you sure you want to queue these changes?
        </h1>
        <section>
          <p>
            You will be changing the metadata of 
            0
             
            works
          </p>
        </section>
        <div
          className="c22"
        >
          <button
            className="c23 dismiss"
            height="41px"
            onClick={[Function]}
          >
            <div
              className="c10 "
              fontFamily={
                Object {
                  "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                  "fontWeight": 500,
                }
              }
              fontSize="14px"
              style={Object {}}
            >
              Go back
            </div>
          </button>
          <button
            className="c17 accept"
            height="41px"
            onClick={[Function]}
          >
            <div
              className="c10 "
              fontFamily={
                Object {
                  "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                  "fontWeight": 500,
                }
              }
              fontSize="14px"
              style={Object {}}
            >
              Continue
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c24"
  />
</div>
`;
